Formula/tidb-server.rb

```diff
@@ -0,0 +1,75 @@
+class TidbServer < Formula
+  desc "TiDB is a MySQL compatible distributed database"
+  homepage "https://www.pingcap.com/en/"
+  url "https://github.com/pingcap/tidb.git",
+      :tag      => "v2.1.5",
+      :revision => "4dc309315c845ee1e1955754325bf2fc212e8f8d"
+
+  depends_on "go" => :build
+
+  def install
+    # Change the default datadir
+    inreplace "config/config.go", "/tmp/tidb", var/"tidb"
+    inreplace "tidb-server/main.go", "/tmp/tidb", var/"tidb"
+
+    # Removing the go.sum file
+    # fixes a problem in go mod hashes
+    rm "go.sum"
```

Comment:
What is the problem with the hashes? Has it been reported, so it gets fixed in the next release?

Reply:
https://stackoverflow.com/questions/54133789/go-modules-checksum-mismatch

TiDB 2.1 upstream will compile with 1.11.2 for its lifetime, so the go.sum reflects how the now invalid hashes were generated.  There is no 1.11.2 in homebrew, which is fine, it will compile on the latest 1.12.x with the go.sum removed.  The upcoming 3.0 release will compile fine without this change, as it will use 1.12.x for its lifetime.

Reply:
@fxcoudert would you like me to link to this stackoverflow thread from the comment? The status of this PR is 'requested changes', but it is not straight forward to me how to make these changes.

Reply:
Right now I'm trying to understand what is the best way to handle things. Maybe for `go.sum` a clearer comment would be nice, but I don't understand why that's a problem for this specific formula and we never see it with other `go` formulas, even when `go` versions change.

Reply:
I assume that other formulas have either not adopted go modules, or have increased the minimum go that they support to avoid 1.11.2.  One could argue that TiDB should be more flexible with which version they pin a GA release to (1.11.x instead of 1.11.2).  But at the rate go changes, that is actually not that easy to do.  As the answerer commented in StackOverflow, unfortunately, breakage is common right now in Go modules (as it was only released in 1.11 and is not yet the default).

Reply:
I have added a link to the StackOverflow thread in https://github.com/Homebrew/homebrew-core/commit/5d7eb8794d5b6756f985ae2ef5d9626a3eeaab75